### PR TITLE
fix: make stkxprt staking respect and update global liquid staking cap

### PR DESCRIPTION
**Pull request by xlab**
## 1. Overview

Replaced direct calls to stakingKeeper with a wrapper that does updates on global liquid stake accounting and checks the caps.

### CODE REVIEW

**Review by kruspy — CHANGES_REQUESTED** [review]
LGTM.

I see any of the `SafelyIncreaseTotalLiquidStakedTokens` and `DecreaseTotalLiquidStakedTokens` messes with the validator specific liquid shares. We really need to make sure of it.

Also, that all entrypoints (Delegate, Undelegate, StakeToLP) are covered by this.

If that is the case, then add tests and we good to go.

**Comment by xlab** [comment]
> should we also add it to the rebalancing (i don't think so), auto compounding (yes)..

autocompounding uses `k.LiquidDelegate` already, so just handling the error better

**Review by puneet2019 — APPROVED** [review]

**Review by kruspy — APPROVED** [review]

File: x/liquidstake/keeper/liquidstake.go [hunk 1]
```diff
@@ -375,6 +375,12 @@ func (k Keeper) LSMDelegate(
 		return sdk.ZeroDec(), sdk.ZeroInt(), types.ErrTooSmallLiquidStakeAmount
 	}
 
+	// after LSM redemption, accounted liquid shares are ignored,
+	// but we have to still account it because now delegation is owned by a LS protocol
+	if err := k.stakingKeeper.SafelyIncreaseTotalLiquidStakedTokens(ctx, stkXPRTMintAmount, false); err != nil {
```
Comment (puneet2019) [inline x/liquidstake/keeper/liquidstake.go:380]: message handler msg Redeem should take care of it? no? just checking if this doesn't double count

Comment (xlab) [reply x/liquidstake/keeper/liquidstake.go:380]: Yeah this is why I wrote comment above - `// after LSM redemption, accounted liquid shares are ignored,`
So basically during LSM we mint shares and then redeem, in order to redelegate stake from user onto protocol. 
Minting increases LS shares, Redeeming decreases LS shares, net is 0. But the delegation is on the LS protocol, so it must be accounted ;)

There is a test for that logic:
https://github.com/persistenceOne/persistenceCore/pull/295/commits/f19026c19d4ba5f0b3dfaca57a364e64ab383b10#diff-728e09d3811928f4460efbdf45c3262de1e346c19fd27fc16be96433e2ffaf72R361

it was not updating the LS counter after LSM redelegation so this is why I added those lines here.

Comment (puneet2019) [reply x/liquidstake/keeper/liquidstake.go:380]: the LS counter should ideally not change after redeem shares (If the redeem is done by a "liquidstaker account".
https://github.com/persistenceOne/cosmos-sdk/blob/v47-lsm/x/staking/keeper/msg_server.go#L873
here is checks if it is liquidstaker account, if it is normal account it reduces, orelse keeps constant

Comment (xlab) [reply x/liquidstake/keeper/liquidstake.go:380]: This was fixed by [https://github.com/persistenceOne/pstake-native/commit/f05ded769667bd5ee43f1e80c8d7d3487739df39#diff-c092ac107fdf7265c82[…]33c9bc6bbd8a79ba6fL42-R45](https://github.com/persistenceOne/pstake-native/commit/f05ded769667bd5ee43f1e80c8d7d3487739df39#diff-c092ac107fdf7265c82873a97c1474690fb4bd8cb0f2f133c9bc6bbd8a79ba6fL42-R45)

**Pull request merged**